src/v/bytes/tests/iobuf_tests.cc

```diff
@@ -568,3 +568,53 @@ SEASTAR_THREAD_TEST_CASE(iobuf_as_istream) {
     BOOST_REQUIRE_EQUAL(underlying.size_bytes(), out.size());
     BOOST_REQUIRE_EQUAL(a + b, out);
 }
+
+SEASTAR_THREAD_TEST_CASE(iobuf_string_view_cmp) {
+    {
+        /// string_view to iobuf comparator, random string
+        auto a = random_generators::gen_alphanum_string(1024);
+        iobuf buf;
+        buf.append(a.data(), a.size());
+        BOOST_REQUIRE_EQUAL(buf, std::string_view(a));
+    }
+    {
+        /// Same length different content
+        static const ss::sstring a = "static_content";
+        static const ss::sstring b = "that_is_not_eq";
+        iobuf buf;
+        buf.append(a.data(), a.size());
+        BOOST_REQUIRE_EQUAL(buf, std::string_view(a));
+        BOOST_REQUIRE_NE(buf, std::string_view(b));
+    }
+    {
+        /// Different length items
+        static const ss::sstring a = "Not the same";
+        static const ss::sstring b = "length";
+        iobuf buf;
+        buf.append(a.data(), a.size());
+        BOOST_REQUIRE_NE(a, std::string_view(b));
+        BOOST_REQUIRE_EQUAL(a, std::string_view(a));
+    }
+    {
+        /// Multiple fragments
+        ss::sstring hello_str;
+        iobuf buf;
+        for (auto i = 0; i < 1000; ++i) {
+            ss::temporary_buffer<char> a("hello", 5);
```

Comment:
I think these are different. one is `h` `e` `l` `l` `o` `\0` and one doens't have the null ptr. 

first you need to assert the strings length are the same.